jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PlatformChecker.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.builder.steps;
+
+import com.google.cloud.tools.jib.api.LogEvent;
+import com.google.cloud.tools.jib.api.buildplan.Platform;
+import com.google.cloud.tools.jib.configuration.BuildContext;
+import com.google.cloud.tools.jib.event.EventHandlers;
+import com.google.cloud.tools.jib.image.json.ContainerConfigurationTemplate;
+import com.google.common.base.Verify;
+import java.util.Set;
+
+/** Provides helper methods to check platforms. */
+public class PlatformChecker {
+
+  /**
+   * Assuming the base image is not a manifest list, checks and warns misconfigured platforms.
+   *
+   * @param buildContext the {@link BuildContext}
+   * @param containerConfig container configuration JSON of the base image
+   */
```

Comment:
Is there a way to check that the build image is in fact not a manifest list in this method, before we proceed to doing the other checks?

Reply:
Yeah, I had the same question. I know this is a bit weird. It's mainly the consequence of factoring out some portion of code to reuse that in multiple places under certain conditions. But I think it's also weird to just pass something unnecessary to the method only to check the argument.

Reply:
Yup, that's a valid point since it's meant to serve as a helper. As long as we check the condition before calling this method, we should be good.